File: yh_check/src/yh_check_distance.cpp
#include "ros/ros.h"
#include "yh_check/YhCheck.h" 



int main(int argc,char**argv)
{
    ros::init(argc,argv, "yh_check_distance");  //노드이름
    ros::NodeHandle nh;  

    ros::Publisher pub = nh.advertise<yh_check::YhCheck>("check_distance",20);  //퍼블리셔 만들기.
   
    
   
    ros::Rate loop_rate(2);  //Hz 주기 만들기.

    yh_check::YhCheck msg;  //메세지 만들기.

    msg.data = true;
    
    


    while(ros::ok())
    {
        msg.stamp = ros::Time::now();  //현재시간
        pub.publish(msg);
        msg.data  = !msg.data; //true면 false, false면 true
        loop_rate.sleep();
    }


    return 0;
}

File: yh_check/scripts/yh_check_distance.py
#!/usr/bin/python             
#-*- coding: utf-8 -*-

import rospy
from yh_check.msg import YhCheck




def distance():
    rospy.init_node("yh_check_distance")
    pub = rospy.Publisher("check_distance",YhCheck, queue_size= 10)

    loop_rate = rospy.Rate(2)  #Hz 넣어주기.

    msg = YhCheck()



    while not rospy.is_shutdown():
       
        msg.stamp = rospy.Time.now()
        msg.data = not msg.data
        rospy.loginfo("send msg : %d", msg.stamp.secs)
        rospy.loginfo("send msg : %d", msg.stamp.nsecs)
        rospy.loginfo("send msg : %d", msg.data)

        pub.publish(msg)
        
        loop_rate.sleep()

        


if __name__ == "__main__":
    try:
        distance()
    except rospy.ROSInterruptException:
            pass